src/github.com/getlantern/flashlight/client/client.go

```diff
@@ -179,3 +191,29 @@ func (client *Client) Configure(cfg *ClientConfig, proxyAll func() bool) {
 func (client *Client) Stop() error {
 	return client.l.Close()
 }
+
+func (client *Client) proxiedDialer(orig func(network, addr string) (net.Conn, error)) func(network, addr string) (net.Conn, error) {
+	var proxied func(network, addr string) (net.Conn, error)
+	if client.ProxyAll() {
+		proxied = orig
+	} else {
+		proxied = detour.Dialer(orig)
+	}
+
+	return func(network, addr string) (net.Conn, error) {
+		if isLanternSpecialDomain(addr) {
+			rewritten := rewriteLanternSpecialDomain(addr)
+			log.Tracef("Rewriting %v to %v", addr, rewritten)
+			return net.Dial(network, rewritten)
+		}
+		return proxied(network, addr)
+	}
+}
+
+func isLanternSpecialDomain(addr string) bool {
+	return strings.Index(addr, LanternSpecialDomainWithColon) == 0
```

Comment:
What's the reason for the constant with a colon? Is this something we could check with https://golang.org/pkg/strings/#HasPrefix and `LanternSpecialDomain`?